shared/crypto/keypair.go

```diff
@@ -35,54 +36,81 @@ type KeyPair interface {
 	Unmarshal([]byte) error
 }
 
-var _ KeyPair = &encKeyPair{}
+var _ KeyPair = &EncKeyPair{}
 
-// encKeyPair struct stores the public key and the passphrase encrypted private key
-type encKeyPair struct {
+// EncKeyPair struct stores the public key and the passphrase encrypted private key
+type EncKeyPair struct {
 	PublicKey     PublicKey
 	PrivKeyArmour string
 }
 
+func (ekp *EncKeyPair) MarshalJSON() ([]byte, error) {
```

Comment:
Just a NIT you should add these to the `KeyPair` interface, use `(kp *encKeyPair)` instead to match the other methods and move them down to be with the other marshalling functions (or replace them if you think this is better) 😅